File: android/app/src/main/java/com/lacour/vincent/wificaresp8266/storage/Preferences.kt
package com.lacour.vincent.wificaresp8266.storage

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager
import com.lacour.vincent.wificaresp8266.constant.Constant
import com.lacour.vincent.wificaresp8266.model.Storage

class Preferences(ctx: Context) {

    private val preferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(ctx)

    private fun getSharedPreferencesValue(storage: Storage): String {
        val (key, default) = storage
        val value = preferences.getString(key, default)!!
        return if (value.isBlank()) default else value
    }

    fun getIpAddress(): String = getSharedPreferencesValue(Constant.IP_ADDRESS_ST0RAGE)
    fun getPort(): String = getSharedPreferencesValue(Constant.PORT_STORAGE)

    fun getMoveForwardValue(): String = getSharedPreferencesValue(Constant.MOVE_FORWARD_STORAGE)
    fun getMoveBackwardValue(): String = getSharedPreferencesValue(Constant.MOVE_BACKWARD_STORAGE)
    fun getStopValue(): String = getSharedPreferencesValue(Constant.MOVE_STOP_STORAGE)
    fun getTurnRightValue(): String = getSharedPreferencesValue(Constant.TURN_RIGHT_STORAGE)
    fun getTurnLeftValue(): String = getSharedPreferencesValue(Constant.TURN_LEFT_STORAGE)

    fun getActionOneValue(): String = getSharedPreferencesValue(Constant.ACTION_1_STORAGE)
    fun getActionTwoValue(): String = getSharedPreferencesValue(Constant.ACTION_2_STORAGE)
    fun getActionThreeValue(): String = getSharedPreferencesValue(Constant.ACTION_3_STORAGE)
    fun getActionFourValue(): String = getSharedPreferencesValue(Constant.ACTION_4_STORAGE)
    fun getActionFiveValue(): String = getSharedPreferencesValue(Constant.ACTION_5_STORAGE)
    fun getActionSixValue(): String = getSharedPreferencesValue(Constant.ACTION_6_STORAGE)
    fun getActionSevenValue(): String = getSharedPreferencesValue(Constant.ACTION_7_STORAGE)
    fun getActionHeightValue(): String = getSharedPreferencesValue(Constant.ACTION_8_STORAGE)

}

File: ios/wificaresp8266/Storage/Preferences.swift
import Foundation

struct Preferences {
    private let defaults = UserDefaults.standard
    
    private func getUserDefaultValue(storage: Storage) -> String {
        let key = storage.key
        let defaultValue = storage.defaultValue
        return defaults.string(forKey: key) ?? defaultValue
    }
    
    func getIpAddress() -> String { getUserDefaultValue(storage: Keys.IP_ADDRESS_ST0RAGE) }
    func getPort() -> String { getUserDefaultValue(storage: Keys.PORT_STORAGE) }
    
    func getMoveForwardValue() -> String { getUserDefaultValue(storage: Keys.MOVE_FORWARD_STORAGE) }
    func getMoveBackwardValue() -> String { getUserDefaultValue(storage: Keys.MOVE_BACKWARD_STORAGE) }
    func getStopValue() -> String { getUserDefaultValue(storage: Keys.MOVE_STOP_STORAGE) }
    func getTurnRightValue() -> String { getUserDefaultValue(storage: Keys.TURN_RIGHT_STORAGE) }
    func getTurnLeftValue() -> String { getUserDefaultValue(storage: Keys.TURN_LEFT_STORAGE) }
    
    func getActionOneValue() -> String { getUserDefaultValue(storage: Keys.ACTION_1_STORAGE) }
    func getActionTwoValue() -> String { getUserDefaultValue(storage: Keys.ACTION_2_STORAGE) }
    func getActionThreeValue() -> String { getUserDefaultValue(storage: Keys.ACTION_3_STORAGE) }
    func getActionFourValue() -> String { getUserDefaultValue(storage: Keys.ACTION_4_STORAGE) }
    func getActionFiveValue() -> String { getUserDefaultValue(storage: Keys.ACTION_5_STORAGE) }
    func getActionSixValue() -> String { getUserDefaultValue(storage: Keys.ACTION_6_STORAGE) }
    func getActionSevenValue() -> String { getUserDefaultValue(storage: Keys.ACTION_7_STORAGE) }
    func getActionHeightValue() -> String { getUserDefaultValue(storage: Keys.ACTION_8_STORAGE) }
}
